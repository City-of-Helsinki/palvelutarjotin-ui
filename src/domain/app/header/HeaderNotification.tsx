import isFuture from 'date-fns/isFuture';
import isPast from 'date-fns/isPast';
import isSameDay from 'date-fns/isSameDay';
import { Notification, NotificationType } from 'hds-react';
import { useTranslation } from 'next-i18next';
import * as React from 'react';
import useLocalStorage from 'react-use/lib/useLocalStorage';

import ExternalLink from '../../../common/components/externalLink/ExternalLink';
import {
  Notification as NotificationNode,
  useNotificationQuery,
} from '../../../generated/graphql-cms';
import useLocale from '../../../hooks/useLocale';
import hash from '../../../utils/hash';
import { useCMSApolloClient } from '../../headless-cms/apollo/apolloClient';

export const NOTIFICATION_STORAGE_KEY = 'header-notification';

type CmsNotificationTypes = 'info' | 'high' | 'low';

// used to get content from inside p-tag (wordpress returns content as html)
const contentRegex = /<p>(.*)<\/p>/i;

type NotificationState = {
  isVisible: boolean;
  closedNotificationHash: string | null | undefined;
};

const notificationTypeMap: Record<CmsNotificationTypes, NotificationType> = {
  info: 'alert',
  high: 'error',
  low: 'info',
};

const HeaderNotification: React.FC = () => {
  const { t } = useTranslation();
  const locale = useLocale();
  const [notificationState, setNotificationState] =
    useLocalStorage<NotificationState>(NOTIFICATION_STORAGE_KEY, {
      isVisible: true,
      closedNotificationHash: null,
    });
  const { isVisible, closedNotificationHash } = notificationState ?? {};
  const cmsApolloClient = useCMSApolloClient();

  const { data } = useNotificationQuery({
    client: cmsApolloClient,
    variables: {
      language: locale ?? 'fi',
    },
    onCompleted: (data) => {
      const notification = data.notification;
      if (notification) {
        const notificationHash = getNotificationHash(notification).toString();
        if (closedNotificationHash !== notificationHash) {
          setNotificationState({
            isVisible: true,
            closedNotificationHash: null,
          });
        }
      }
    },
  });

  const notification = data?.notification;
  const parsedContent = contentRegex.exec(notification?.content ?? '')?.[1];
  const { title, level, linkText, linkUrl } = notification ?? {};

  const handleCloseNotification = () => {
    setNotificationState({
      isVisible: false,
      closedNotificationHash: notification
        ? getNotificationHash(notification).toString()
        : null,
    });
  };

  return isNotificationActive(notification) && isVisible ? (
    <Notification
      type={notificationTypeMap[(level as CmsNotificationTypes) ?? 'info']}
      label={title ?? undefined}
      dismissible={true}
      closeButtonLabelText={
        t('header:notification:labelCloseNotification') as string
      }
      onClose={handleCloseNotification}
    >
      {parsedContent}
      {linkUrl && (
        <>
          <ExternalLink
            href={linkUrl}
            style={{ marginTop: '1rem', display: 'block' }}
          >
            {linkText}
          </ExternalLink>
        </>
      )}
    </Notification>
  ) : null;
};

const getNotificationHash = (notification: NotificationNode) => {
  // notification fields used to build string for unique hash
  const stringKeys = [
    'title',
    'content',
    'linkText',
    'linkUrl',
    'level',
    'endDate',
    'startDate',
  ] as const;
  const combinedString = stringKeys.reduce(
    (combinedString, current) =>
      (combinedString ?? '') + (notification[current] ?? ''),
    ''
  );
  return hash(combinedString);
};

const isNotificationActive = (notification?: NotificationNode | null) => {
  const { startDate, endDate, title, content } = notification ?? {};
  const isNotificationContentMissing = !content && !title;

  if (!notification || isNotificationContentMissing) return false;

  const isSameDayOrFuture = (date: Date) =>
    isSameDay(new Date(), date) || isFuture(date);

  return [
    // both start and end dates are defined
    () =>
      startDate &&
      endDate &&
      isPast(new Date(startDate)) &&
      isSameDayOrFuture(new Date(endDate)),
    // only end date defined
    () => !startDate && endDate && isSameDayOrFuture(new Date(endDate)),
    // only start date defined
    () => startDate && !endDate && isPast(new Date(startDate)),
    // neither start or end date are defined
    () => !startDate && !endDate,
  ].some((f) => !!f());
};

export default HeaderNotification;
