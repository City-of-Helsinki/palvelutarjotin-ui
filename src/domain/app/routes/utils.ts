import { CollectionItemType } from 'react-helsinki-headless-cms';

import { ROUTES } from './constants';

export const getCmsPagePath = (uri?: string | null): string => {
  return ROUTES.CMS_PAGE.replace('/:uri', uri ?? '');
};

export const getCmsArticlePath = (uri?: string | null): string => {
  return ROUTES.CMS_ARTICLE.replace('/:uri', uri ?? '');
};

export const getCollectionItemUrl = (item: CollectionItemType): string => {
  if (!item) {
    return '#';
  }
  if (item.__typename === 'Post') {
    return getCmsArticlePath(item.uri);
  }
  if (item.__typename === 'Page') {
    return getCmsPagePath(item.uri);
  }
  return item?.uri ?? '';
};
