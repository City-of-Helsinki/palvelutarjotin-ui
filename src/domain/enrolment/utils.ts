import {
  EnrolOccurrenceMutationInput,
  Language,
  NotificationType,
} from '../../generated/graphql';
import { EnrolmentFormFields } from './enrolmentForm/constants';

const getNotificationType = (values: EnrolmentFormFields): NotificationType => {
  return values.hasEmailNotification && values.hasSmsNotification
    ? NotificationType.EmailSms
    : NotificationType.Email;
};

export const getEnrolmentPayload = ({
  occurrenceIds,
  values,
}: {
  occurrenceIds: string[];
  values: EnrolmentFormFields;
}): EnrolOccurrenceMutationInput => {
  return {
    occurrenceIds,
    notificationType: getNotificationType(values),
    person: values.isSameResponsiblePerson
      ? undefined
      : { ...values.person, language: values.language as Language },
    studyGroup: {
      ...values.studyGroup,
      amountOfAdult: Number(values.studyGroup.amountOfAdult),
      groupSize: Number(values.studyGroup.groupSize),
      person: {
        ...values.studyGroup.person,
        language: values.language as Language,
      },
      studyLevels: values.studyGroup.studyLevels,
    },
  };
};

declare let grecaptcha: ReCAPTCHA;

interface ReCAPTCHA {
  ready: (f: () => void) => void;
  execute: (key: string, action: { action: string }) => Promise<string>;
}

export const getCAPTCHAToken = (): Promise<string> => {
  return new Promise((resolve, reject) => {
    if (process.env.NEXT_PUBLIC_CAPTCHA_KEY) {
      const captchaKey = process.env.NEXT_PUBLIC_CAPTCHA_KEY;
      grecaptcha.ready(() => {
        grecaptcha
          .execute(captchaKey, { action: 'submit' })
          .then(resolve)
          .catch(reject);
      });
    } else {
      reject('CAPTCHA_KEY missing!');
    }
  });
};
