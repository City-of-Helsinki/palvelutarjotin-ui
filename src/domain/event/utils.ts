import { isToday, isTomorrow } from 'date-fns';
import { TFunction } from 'next-i18next';

import { EventFieldsFragment } from '../../generated/graphql';
import { Language } from '../../types';
import formatDate from '../../utils/formatDate';
import getLocalisedString from '../../utils/getLocalisedString';
import getTimeFormat from '../../utils/getTimeFormat';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';

/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

export const getEventStartTimeStr = (
  event: EventFieldsFragment,
  locale: Language,
  t: TFunction
): string | null => {
  const startTime = event.startTime ? new Date(event.startTime) : null;
  const timeFormat = getTimeFormat(locale);

  if (!startTime) return null;

  if (isToday(startTime))
    return t('event:eventCard.startTime.today', {
      time: formatDate(startTime, timeFormat, locale),
    });

  if (isTomorrow(startTime))
    return t('event:eventCard.startTime.tomorrow', {
      time: formatDate(startTime, timeFormat, locale),
    });

  return t('event:eventCard.startTime.other', {
    date: formatDate(startTime),
    time: formatDate(startTime, timeFormat, locale),
  });
};

/**
 * Check is event free
 * @param eventData
 * @return {boolean}
 */
export const isEventFree = (event: EventFieldsFragment): boolean =>
  Boolean(event.offers.find((item) => item.isFree)?.isFree);

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const getEventFields = (
  event: EventFieldsFragment | undefined | null,
  locale: Language
) =>
  event
    ? {
        isEventFree: isEventFree(event),
        eventName: getLocalisedString(event?.name, locale),
        shortDescription: getLocalisedString(event?.shortDescription, locale),
        description: getLocalisedString(event?.description, locale),
        imageUrl: event?.images?.[0]?.url,
        imageAltText: event?.images?.[0]?.altText,
        photographerName: event?.images?.[0]?.photographerName,
        organization: event.pEvent?.organisation?.name,
        contactPhoneNumber: event.pEvent?.contactPhoneNumber,
        contactEmail: event.pEvent?.contactEmail,
        contactPerson: event.pEvent?.contactPerson?.name,
      }
    : {};
