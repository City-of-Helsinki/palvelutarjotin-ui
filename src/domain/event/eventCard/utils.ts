import { TFunction } from 'next-i18next';

import { EnrolmentStatusKeywordPropsType } from '../../../common/components/keyword/EnrolmentStatusKeyword';
import { EventsFieldsFragment } from '../../../generated/graphql';
import { Language } from '../../../types';
import { formatDateRange } from '../../../utils/time/format';
import { isMultidayOccurrence } from '../../occurrence/utils';
import { getEventStartTimeStr } from '../utils';

export const getOccurrenceDateStr = (
  occurrence: { startTime: string; endTime: string },
  locale: Language,
  t: TFunction
): string | null => {
  if (isMultidayOccurrence(occurrence)) {
    return formatDateRange(
      new Date(occurrence.startTime),
      new Date(occurrence.endTime)
    );
  }

  return getEventStartTimeStr(new Date(occurrence.startTime), locale, t);
};

/**
 * Has the event enrolments enabled in any of it's occurrences?
 * @param event The event to check the status for.
 * @returns `true` if any of the event occurrences has enrolments and
 *  `false` if none.
 */
export const hasEventEnrolmentsEnabled = (event: EventsFieldsFragment) =>
  Boolean(event.pEvent.enrolmentStart) || hasEventExternalEnrolments(event);

/**
 * Has the event external enrolments enabled in any of it's occurrences?
 *
 * @param event The event to check the status for.
 * @returns `true` if the event has some external enrolments and
 *  `false` if none of the occurrences has external enrolments.
 */
export const hasEventExternalEnrolments = (event: EventsFieldsFragment) =>
  Boolean(event.pEvent.externalEnrolmentUrl);

/**
 * Resolve the enrolment status of an event. Has the event free spots available?
 *
 * @param event The event to check the status for.
 * @returns Returns `null` if the enrolment status could not be resolved or
 *    the event has no enrolments at all,
 *    `true` if there are any free spots available and `false` if there are no
 *    spots available.
 */
export const hasEventFreeSpotsAvailable = (
  event: EventsFieldsFragment
): boolean | null => event.pEvent.hasSpaceForEnrolments ?? null;

/**
 * Determines the enrollment status of an event.
 *
 * @param event The event to check the enrollment status for.
 * @returns The enrollment status of the event, or `undefined`
 *      if no enrolments exist or the status could not be determined.
 */
export const getEventEnrolmentStatus = (
  event: EventsFieldsFragment
): EnrolmentStatusKeywordPropsType['enrolmentStatus'] => {
  if (hasEventEnrolmentsEnabled(event)) {
    if (hasEventExternalEnrolments(event)) {
      return 'externally_unknown';
    }
    const hasEventFreeSpots = hasEventFreeSpotsAvailable(event);

    if (hasEventFreeSpots === false) {
      // Event has no free spots available, so it is full.
      return 'full';
    } else if (hasEventFreeSpots === true) {
      // Event has some free spots available.
      return 'free_space_available';
    }
  }
  // No enrolments at all or the enrolment status could not be resolved
  return undefined;
};
