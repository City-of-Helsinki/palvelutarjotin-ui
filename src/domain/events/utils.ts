import isValidDate from 'date-fns/isValid';

import { EVENT_LANGUAGES } from '../../constants';
import {
  EventsQueryVariables,
  EventFieldsFragment,
  EventsQuery,
} from '../../generated/graphql';
import { queryParameterToArray } from '../../utils/queryParameterToArray';
import { EventSearchFormValues } from './eventSearchForm/EventSearchForm';

export const getTextFromDict = (
  query: NodeJS.Dict<string | string[]>,
  key: string,
  defaultValue = ''
): string | undefined => {
  const value = query[key];
  if (Array.isArray(value)) return value.join(',');
  return (value as string) || defaultValue;
};

type EventFilterOptions = {
  page?: number;
  pageSize?: number;
};

export const getEventFilterVariables = (
  query: NodeJS.Dict<string | string[]>,
  options?: EventFilterOptions
): EventsQueryVariables => ({
  include: ['keywords,location'],
  text: getTextFromDict(query, 'text', undefined),
  inLanguage: getTextFromDict(query, 'inLanguage', undefined),
  start:
    typeof query.date === 'string' && isValidDate(new Date(query.date))
      ? new Date(query.date).toISOString()
      : null,
  ...options,
});

export const getInitialValues = (
  query: NodeJS.Dict<string | string[]>
): EventSearchFormValues => {
  return {
    text: getTextFromDict(query, 'text') || '',
    date: getInitialDate(query.date),
    inLanguage: queryParameterToArray(query.inLanguage as EVENT_LANGUAGES),
  };
};

export const getInitialDate = (date?: string | string[]): Date | null => {
  if (typeof date === 'string' && isValidDate(new Date(date))) {
    return new Date(date);
  }

  return null;
};

export const getEventsThatHaveUpcomingOccurrence = (
  eventData?: EventsQuery
): EventFieldsFragment[] | undefined => {
  return eventData?.events?.data.filter((e) => e.pEvent.nextOccurrenceDatetime);
};
