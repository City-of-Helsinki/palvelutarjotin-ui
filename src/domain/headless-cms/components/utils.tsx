import { randomUUID } from 'crypto';

import React from 'react';
import {
  ArticleType,
  Card,
  Collection,
  GeneralCollectionType,
  getCollectionCards,
  PageType,
} from 'react-helsinki-headless-cms';

import { getCmsArticlePath, getCmsPagePath } from '../../app/routes/utils';

export function getCmsCollectionList(
  collections: GeneralCollectionType[]
): React.ReactElement<typeof Collection>[] {
  return collections.map((collection) => (
    <Collection
      key={`collection-${randomUUID()}`}
      title={collection.title}
      collectionContainerProps={{
        withDots: collection.items.length < 4 ? false : true,
      }}
      type="grid"
      cards={getCollectionCards(collection as GeneralCollectionType, []).map(
        (item) => (
          <Card
            key={`card-${item.id}`}
            {...item}
            clampText={true}
            withShadow={true}
            hasLink={true}
            url={getCollectionItemUrl(item as PageType)}
            // imageLabel={item.featuredImage?.node?.title ?? ''}
            // imageUrl={item.featuredImage?.node?.mediaItemUrl ?? ''}
          />
        )
      )}
    />
  ));
}

export function getCollectionItemUrl(item?: PageType | ArticleType): string {
  if (!item) {
    return '#';
  }
  if (item.__typename === 'Post') {
    return getCmsArticlePath(item.uri);
  }
  if (item.__typename === 'Page') {
    return getCmsPagePath(item.uri);
  }
  return item?.uri ?? '';
}
