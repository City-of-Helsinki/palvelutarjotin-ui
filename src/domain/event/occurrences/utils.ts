import {
  EventFieldsFragment,
  OccurrenceFieldsFragment,
} from '../../../generated/graphql';
import { ENROLMENT_ERRORS } from '../../enrolment/constants';
import {
  hasOccurrenceSpace,
  isEnrolmentClosed,
  isOccurrenceCancelled,
} from '../../occurrence/utils';

export function getEnrolmentError(
  occurrence: OccurrenceFieldsFragment,
  event: EventFieldsFragment
): ENROLMENT_ERRORS | null {
  if (isEnrolmentClosed(occurrence, event))
    return ENROLMENT_ERRORS.ENROLMENT_CLOSED_ERROR;
  if (!hasOccurrenceSpace(occurrence))
    return ENROLMENT_ERRORS.NOT_ENOUGH_CAPACITY_ERROR;
  if (isOccurrenceCancelled(occurrence)) {
    return ENROLMENT_ERRORS.ENROLMENT_CANCDELLED_ERROR;
  }

  return null;
}

export function getOrderedLanguages(
  languages: { id: string; name: string }[]
): string[] {
  const languageOrder = ['fi', 'sv', 'en'];
  const languageStrings = languages.map((l) => l.id.split('_')[0]);

  // order languages based on languageOrder, other languages go
  // to the end of the array
  return languageStrings.sort((a, b) => {
    const aIndex = languageOrder.indexOf(a);
    const bIndex = languageOrder.indexOf(b);

    if (aIndex > -1 && bIndex > -1) {
      return aIndex - bIndex;
    }

    if (aIndex > -1 && bIndex === -1) {
      return -1;
    }

    if (bIndex > -1 && aIndex === -1) {
      return 1;
    }

    return 0;
  });
}
