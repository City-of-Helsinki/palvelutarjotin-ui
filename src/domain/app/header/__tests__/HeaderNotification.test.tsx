import { advanceTo } from 'jest-date-mock';
import * as React from 'react';

import {
  Notification,
  NotificationDocument,
} from '../../../../generated/graphql-cms';
import { fakeNotification } from '../../../../utils/cmsMockDataUtils';
import {
  render,
  screen,
  userEvent,
  waitFor,
} from '../../../../utils/testUtils';
import HeaderNotification, {
  NOTIFICATION_STORAGE_KEY,
} from '../HeaderNotification';

const notificationTitle = 'Notification title';
const notificationContent = 'Notification content';

const renderComponent = (notification: Partial<Notification> = {}) => {
  return render(<HeaderNotification />, {
    mocks: [
      {
        request: {
          query: NotificationDocument,
          variables: {
            language: 'fi',
          },
        },
        result: {
          data: {
            notification: fakeNotification({
              title: notificationTitle,
              content: `<p>${notificationContent}</p>`,
              ...notification,
            }),
          },
        },
      },
    ],
  });
};

beforeEach(() => {
  advanceTo(new Date(2020, 7, 10));
});

it('matches snapshot', async () => {
  const { container } = renderComponent({
    linkUrl: 'https://kultus.hel.fi',
    level: 'info',
    title: 'Notification title',
    content: 'Notification content',
    linkText: 'link text',
  });

  await screen.findByRole('heading', { name: 'Notification title' });
  expect(container).toMatchSnapshot();
});

it('renders notification data correctly', async () => {
  const title = 'Notification title';
  const content = 'Notification content';
  const linkUrl = 'https://kultus.hel.fi';
  const linkText = 'Linkki sivustolle';
  renderComponent({
    startDate: new Date(2020, 6, 10).toISOString(),
    endDate: new Date(2020, 10, 10).toISOString(),
    title,
    content: `<p>${content}</p>`,
    linkUrl,
    linkText,
  });

  await screen.findByText(title);
  await screen.findByText(content);
  await screen.findByRole('link', {
    name: /Linkki sivustolle avautuu uudessa välilehdessä/i,
  });

  await userEvent.click(
    await screen.findByRole('button', {
      name: 'Sulje huomiotiedote',
    })
  );

  await waitFor(() => {
    expect(screen.queryByText(content)).not.toBeInTheDocument();
  });
});

it('saves notification state to local storage', async () => {
  renderComponent();

  await screen.findByText(notificationContent);

  await waitFor(() => {
    expect(
      JSON.parse(localStorage.getItem(NOTIFICATION_STORAGE_KEY) ?? '')
    ).toMatchObject({ isVisible: true, closedNotificationHash: null });
  });

  await userEvent.click(
    screen.getByRole('button', {
      name: 'Sulje huomiotiedote',
    })
  );

  await waitFor(() => {
    expect(
      JSON.parse(localStorage.getItem(NOTIFICATION_STORAGE_KEY) ?? '')
    ).not.toMatchObject({ isVisible: true, closedNotificationHash: null });
  });

  const localStorageObjectAfterClosing = JSON.parse(
    localStorage.getItem(NOTIFICATION_STORAGE_KEY) ?? ''
  );

  expect(localStorageObjectAfterClosing).toMatchInlineSnapshot(`
    {
      "closedNotificationHash": "6176514228182135",
      "isVisible": false,
    }
  `);

  await waitFor(() => {
    expect(screen.queryByText(notificationContent)).not.toBeInTheDocument();
  });
});

it('render notification even if title is missing', async () => {
  renderComponent({
    title: '',
  });

  await screen.findByText(notificationContent);
});

it('renders error-level notification if level is high', async () => {
  renderComponent({
    level: 'high',
  });

  const notification = await screen.findByRole('region', {
    name: /notification/i,
  });

  expect(notification.className).toContain('Notification-module_error');
});

it('renders default notification if level is low', async () => {
  renderComponent({
    level: 'low',
  });

  const notification = await screen.findByRole('region', {
    name: /notification/i,
  });

  expect(notification.className).toContain('Notification-module_notification');
  expect(notification.className).not.toContain('Notification-module_error');
  expect(notification.className).not.toContain('Notification-module_alert');
});

it('renders alert-level notification if level is info', async () => {
  renderComponent({
    level: 'info',
  });

  const notification = await screen.findByRole('region', {
    name: /notification/i,
  });

  expect(notification.className).toContain('Notification-module_alert');
});

it('renders notification if endDate is same day', async () => {
  const title = 'Notification title';
  renderComponent({
    endDate: new Date(2020, 7, 10).toISOString(),
    title,
  });

  await screen.findByText(title);
});

it('renders notification if endDate is in the future', async () => {
  const title = 'Notification title';
  renderComponent({
    endDate: new Date(2020, 10, 10).toISOString(),
    title,
  });

  await screen.findByText(title);
});

it("doesn't render notification if startDate is in the future", async () => {
  const title = 'Notification title';
  renderComponent({
    startDate: new Date(2020, 10, 10).toISOString(),
    title,
  });

  await waitFor(() => {
    expect(screen.queryByText(title)).not.toBeInTheDocument();
  });
});

it("doesn't render notification if endDate is in the past", async () => {
  const title = 'Notification title';
  renderComponent({
    endDate: new Date(2020, 6, 10).toISOString(),
    title,
  });

  await waitFor(() => {
    expect(screen.queryByText(title)).not.toBeInTheDocument();
  });
});

it('renders notification if startDate is in the past and endDate is in the future', async () => {
  const title = 'Notification title';
  renderComponent({
    endDate: new Date(2020, 10, 10).toISOString(),
    startDate: new Date(2020, 6, 10).toISOString(),
    title,
  });

  await screen.findByText(title);
});
