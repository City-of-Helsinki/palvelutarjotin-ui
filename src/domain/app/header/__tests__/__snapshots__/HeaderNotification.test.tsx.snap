// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <section
    aria-label="Notification"
    class="Notification-module_notification__2hN3H notification_hds-notification__2DQPN Notification-module_alert__1Erry notification_hds-notification--alert__1MV7H"
  >
    <div
      class="Notification-module_content__ZZKvE notification_hds-notification__content__5ylSS"
    >
      <div
        aria-level="2"
        class="Notification-module_label__1bhNu notification_hds-notification__label__2t1lR"
        role="heading"
      >
        <svg
          aria-hidden="true"
          aria-label="alert-circle-fill"
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Notification-module_icon__1ans7 notification_hds-icon__29HQM"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 3C16.9706 3 21 7.02944 21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12C3 7.02944 7.02944 3 12 3ZM13 16V18H11V16H13ZM13 6V14H11V6H13Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        Notification title
      </div>
      <div
        class="Notification-module_body__9BSAD notification_hds-notification__body__3cobH"
      >
        <a
          href="https://kultus.fi"
          rel="noreferrer"
          style="margin-top: 1rem; display: block;"
          target="_blank"
        >
          link text
          <svg
            aria-hidden="true"
            aria-label="link-external"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            style="vertical-align: middle; margin-left: 0.5rem;"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5 3V19H21V21H3V3H5ZM21 3V12H19V6.413L9.91421 15.5L8.5 14.0858L17.585 5H12V3H21Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="srOnly"
          >
            Avautuu uudessa välilehdessä
          </div>
        </a>
      </div>
    </div>
    <button
      aria-label="Sulje huomiotiedote"
      class="Notification-module_close__JWYav notification_hds-notification__close-button__1yGIj Notification-module_alert__1Erry notification_hds-notification--alert__1MV7H"
      title="Sulje huomiotiedote"
      type="button"
    >
      <svg
        aria-hidden="true"
        aria-label="cross"
        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M18 7.5L13.5 12L18 16.5L16.5 18L12 13.5L7.5 18L6 16.5L10.5 12L6 7.5L7.5 6L12 10.5L16.5 6L18 7.5Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </section>
</div>
`;
